src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -82,6 +90,40 @@ public ReadOnlyAddressBook getAddressBook() {
         return favouritedList;
     }
 
+    /**
+     * Iterate through the list of Persons and return number of persons in
+     * the region given
+     * @param region is the region where buyers have their preference
+     *               for their potential property and sellers have their property in
+     */
+    @Override
+    public int getPersonsBasedOnRegion(String region) {
+        int totalPersons = 0;
+        //Resets to full list of Persons to prevent any logical error after `find` command
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        //@flairekq
+        //Author provided code to help with arrow code issue: https://github.com/nus-cs2103-AY2122S2/forum/issues/233
+        if (!Region.isValidRegion(region)) { //defensive code
+            return totalPersons;
+        }
+        for (Person person : getFilteredPersonList()) {
+            Preference preference = person.getPreference().isPresent() ? person.getPreference().get() : null;
+            if (person.getUserType().isBuyer() && preference != null && preference.getRegionInString().equals(region)) {
+                totalPersons++;
+                continue;
+            }
+            //If usertype is seller
+            Set<Property> setOfPropertyValues = person.getProperties();
+            for (Property p : setOfPropertyValues) {
+                if (p.getRegionInString().equals(region)) {
+                    totalPersons++;
+                }
+            }
+        }
+        return totalPersons;
+        //@flairekq
+    }
```

Comment:
Not really a big issue, but I feel like it would make more sense to convert the input `String` into a `Region` object instead for comparison.

This will let you remove the `getRegionInString()` method which is duplicated in both `Property` and `Preference`.